### ISSUE TITLE
Black list more origins

### ISSUE THREAD

**Issue by ceckoslab** [davidjbradshaw/iframe-resizer#771]
Hello Iframe Resizers,

I am working on a project called Boomerang JS which is a performance tracking JS library - https://github.com/akamai/boomerang . For our clients we include our library in special Iframe for performance reasons.

This iframe doesn't point to real url but we have 3 variations where it points to:

- `<iframe src="about:blank" ...`
- `<iframe src="javascript:void(0)" ...`
- `<iframe src="..." ...`

We noticed that this is causing a problem when Iframe Resizer is calling **postMessage()**:
```
VM247:1 Uncaught DOMException: Failed to execute 'postMessage' on 'Window': Invalid target origin 'about:blank' in a call to 'postMessage'.
    at <anonymous>:1:8
```

We instructed our clients to try `checkOrigin: false` or `checkOrigin: ["https://www.example.com"]` in order to resolve the problem but we would try to make things smooth in the future.

I see that Iframe Resizer is already a check for `file://` here:
```
function getTargetOrigin(remoteHost) {
    return '' === remoteHost || 'file://' === remoteHost ? '*' : remoteHost
}
```

Do you think that we can add at least `about:blank` and `javascript:void(0)` check in order to black list those "Origins"? If yes I will be happy to contribute with a pull request.

**Comment by davidjbradshaw** [comment]
This fails, as no page has been loaded into the iframe, this is an issue with Chrome, you will need to raise a bug with Google to address this.

Alternatively you could create your own blank HTML page to use instead

**Comment by nicjansma** [comment]
@davidjbradshaw Boomerang `document.write`'s a real "page" into the frame (which is where Boomerang runs from).

(more details @ https://akamai.github.io/boomerang/tutorial-loader-snippet.html)

Seems if `file:` is already being blacklisted, `about:` and `javascript:` schemas could be as well.

**Comment by davidjbradshaw** [comment]
Oh I think I better understand what your asking now, happy to take a PR for that

**Comment by ceckoslab** [comment]
Hello @davidjbradshaw 

I am currently working on pull request. I did some code changes and created a test where we inject on the page iframes with src "about:blank" and  "javascript:void(0)".

How do you like this:

```
function getTargetOrigin(remoteHost) {
    var re = new RegExp('^(about:blank|javascript:)')

    if (null !== remoteHost.match(re)) {
      return '*'
    }

    return '' === remoteHost || 'file://' === remoteHost ? '*' : remoteHost
}
```

I didn't want to modify the part where we have `'file://' === remoteHost` because I see that we check if remoteHost is exactly 'file://' but if you want I can modify the code in a way that we will keep everything compact on one line:

```
function getTargetOrigin(remoteHost) {
  return '' === remoteHost || null !== remoteHost.match(/^(about:blank|javascript:|file:\/\/)/) ? '*' : remoteHost
}
```

Please let me know what do you think.

**Comment by davidjbradshaw** [comment]
Looks good to me

**Comment by ceckoslab** [comment]
Hello @davidjbradshaw 

I created a Pull Request. Also before I pushed I did run:
```
grunt build
grunt test
```

There is a test file under: `test/noMessageForBlackListedOrigins.html`.

Before fix the test result was:
```
Testing test/noMessageForBlackListedOrigins.html ..
>> PhantomJS timed out, possibly due to:
>> - QUnit is not loaded correctly.
>> - A missing QUnit start() call.
>> - Or, a misconfiguration of this task.
```

After fix the test result is:
```
Testing test/noMessageForBlackListedOrigins.html ..OK
```

Let me know if I can support more :)

**Comment by davidjbradshaw** [comment]
Thanks for fixing the test as well. Help is always welcome, if you want to look at the outstanding issues :)

**Comment by davidjbradshaw** [comment]
Released as v4.2.8
